tests/components/yeelight/test_init.py

```diff
@@ -67,3 +75,57 @@ async def test_setup_import(hass: HomeAssistant):
     assert hass.states.get(f"binary_sensor.{name}_nightlight") is not None
     assert hass.states.get(f"light.{name}") is not None
     assert hass.states.get(f"light.{name}_nightlight") is not None
+
+
+async def test_unique_ids_device(hass: HomeAssistant):
+    """Test Yeelight unique IDs from yeelight device IDs."""
+    config_entry = MockConfigEntry(
+        domain=DOMAIN,
+        data={
+            **CONFIG_ENTRY_DATA,
+            CONF_NIGHTLIGHT_SWITCH: True,
+        },
+        unique_id=ID,
+    )
+    config_entry.add_to_hass(hass)
+
+    mocked_bulb = _mocked_bulb()
+    mocked_bulb.bulb_type = BulbType.WhiteTempMood
+    with _patch_discovery(MODULE), patch(f"{MODULE}.Bulb", return_value=mocked_bulb):
+        assert await hass.config_entries.async_setup(config_entry.entry_id)
+        await hass.async_block_till_done()
+
+    er = await entity_registry.async_get_registry(hass)
+    assert er.async_get(ENTITY_BINARY_SENSOR).unique_id == ID
+    assert er.async_get(ENTITY_LIGHT).unique_id == ID
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6372789)